apps/crn-server/test/integration/common/reminders.integration-test.ts

```diff
@@ -0,0 +1,580 @@
+
+    beforeAll(async () => {
+      event = await createEvent({
+        startDate: new Date('2022-08-10T16:00:00.0Z').toISOString(),
+        endDate: new Date('2022-08-11T16:00:00.0Z').toISOString(),
+      });
+    });
+
+    afterAll(async () => {
+      await fixtures.deleteEvents([event.id]);
+    });
+
+    test('Should see the reminder when the event starts today', async () => {
+      // setting system time to 5AM in UTC, event happening at 4PM in UTC
+      jest.setSystemTime(new Date('2022-08-10T05:00:00.0Z'));
+      await expectReminderWithId(`event-happening-today-${event.id}`);
+    });
+
+    test('Should not see the reminder if the event has already started', async () => {
+      // setting system time to 5PM in UTC, event happening at 4PM in UTC
+      jest.setSystemTime(new Date('2022-08-10T17:00:00.0Z'));
+      await expectNotToContainReminderWithId(
+        `event-happening-today-${event.id}`,
+      );
+    });
+
+    test("Should not see the reminder if the event is happening on the next day of the user's timezone", async () => {
+      // setting system time to 5AM in UTC, event happening at 4PM in UTC
+      jest.setSystemTime(new Date('2022-08-10T05:00:00.0Z'));
+
+      // requesting reminders for the user based in LA where 5AM UTC is 10PM the previous day
+      const timezone = 'America/Los_Angeles';
+
+      await expectNotToContainReminderWithId(
+        `event-happening-today-${event.id}`,
+        app,
+        timezone,
+      );
+    });
+  });
+
+  describe('Event Happening Now Reminder', () => {
+    let event: EventFixture;
+
+    beforeAll(async () => {
+      event = await createEvent({
+        startDate: new Date('2022-08-10T10:00:00.0Z').toISOString(),
+        endDate: new Date('2022-08-10T11:00:00.0Z').toISOString(),
+      });
+    });
+
+    afterAll(async () => {
+      await fixtures.deleteEvents([event.id]);
+    });
+
+    test('Should see the reminder when the event has started but has not finished', async () => {
+      // setting system time to 10:05AM in UTC
+      // event happening at 10PM in UTC and ending at 11AM UTC
+      jest.setSystemTime(new Date('2022-08-10T10:05:00.0Z'));
+      await expectReminderWithId(`event-happening-now-${event.id}`);
+    });
+
+    test('Should not see the reminder when the event has already ended', async () => {
+      // setting system time to 11:05AM in UTC
+      // event happening at 10PM in UTC and ending at 11AM UTC
+      jest.setSystemTime(new Date('2022-08-10T11:05:00.0Z'));
+      await expectNotToContainReminderWithId(`event-happening-now-${event.id}`);
+    });
+
+    test('Should not see the reminder when the event is happening but its status is Draft', async () => {
+      // setting system time to 10:05AM in UTC
+      // event happening at 10PM in UTC and ending at 11AM UTC
+      jest.setSystemTime(new Date('2022-08-10T10:05:00.0Z'));
+      await fixtures.publishEvent(event.id, 'Draft');
+      await delay(1000);
```

Comment:
This is to give time for the cache to clear so that the negative assertion doesn't trivially pass, right? It might be worth just adding a comment to explain it because it looks redundant at first glance.

Reply:
actually I've added this because this tests was failing randomly, I believe it fails sometimes because it of the time it takes to convert the event to draft

Reply:
I've added a comment here to explain that [add comment explain delay inside test](https://github.com/yldio/asap-hub/pull/3236/commits/9c1687ecd9c0e459cabfc134456e7697627e6993) :)